lxd/operations/operations.go

```diff
@@ -269,6 +283,10 @@ func (op *Operation) done() {
 			// So don't log warning if operation not found.
 			op.logger.Warn("Failed to delete operation", logger.Ctx{"status": op.status, "err": err})
 		}
+
+		if op.onDone != nil {
```

Comment:
Why did you add op.opDone rather than just checking `op.requestor` is non-nil and extracting the callback from the request's context if present here?

Reply:
This is because the request context is not be acessible from here, just from `OperationCreate`.
I could add `op.request` instead of `op.onDone` to make the context acessible from `done()` but I don't see how that is better.

Reply:
Ah i thought SetRequestore stored r